theworkoutinfo/ViewControllers/HomeViewController.swift

```diff
@@ -57,22 +67,51 @@ class HomeViewController: UIViewController {
         self.tableView.es.addInfiniteScrolling {
             [unowned self] in
             self.pageNumber += 1
-            self.viewModel.loadExercises.onNext(self.pageNumber)
+            self.isCategorySelected ? self.viewModel.loadExercisesByCategory.onNext((false, self.categorySelected!, self.pageNumber)) : self.viewModel.loadExercises.onNext((false, self.pageNumber))
         }
+        
+        categoryTextField.inputView = categoryPickerView
+        categoryTextField.text = "Select a category"
+        filterView.backgroundColor = UIColor.greenAppColor()
     }
     
     func setupBindings() {
+        viewModel.loadExercises.onNext((false,1))
+        viewModel.loadCategories.onNext(())
+        
         viewModel.exercises.asObservable().subscribe(onNext: { _ in
             self.rc.endRefreshing()
             self.tableView.es.stopLoadingMore()
-            print("It Worked!")
         }).disposed(by: disposeBag)
         
         rc.rx.controlEvent(.valueChanged).subscribe({ _ in
-            self.viewModel.loadExercises.onNext(1)
+            self.pageNumber = 1
+            self.isCategorySelected ? self.viewModel.loadExercisesByCategory.onNext((false, self.categorySelected! ,self.pageNumber)) : self.viewModel.loadExercises.onNext((false, 1))
```

Comment:
the VC shouldn't be the one to send the category to the VM, the viewModel should already know it